logger/unilog.go

```diff
@@ -374,16 +455,18 @@ func (u *Unilog) Main() {
 	u.target = flag.Arg(0)
 	u.reopen()
 
-	u.lines, u.errs = readlines(os.Stdin, u.BufferLines, u.shutdown)
-
 	fileName := u.target
 
 	Stats = setupStatsd(u.StatsdAddress, fileName, statstags)
 	Stats.Tags = append(Stats.Tags, fmt.Sprintf("file_name:%s", fileName))
 
 	clevels.Stats = setupStatsd(u.StatsdAddress, fileName, cleveltags)
 
+	tagState = setupIndependentTags()
```

Comment:
Let's move this above where we initialize the stats clients, to guard against new code that emits metrics coming between "we've got a stats client" and "we've got the tags initialized"

Reply:
👍